src/controller/SetUpCodePairer.cpp

```diff
@@ -398,6 +419,11 @@ void SetUpCodePairer::OnDiscoveredDeviceOverWifiPAF()
 
 void SetUpCodePairer::OnWifiPAFDiscoveryError(CHIP_ERROR err)
 {
+    if (mWaitingForDiscovery[kWiFiPAFTransport] == false)
```

Comment:
Again, if discovery has stopped this object may have been destroyed and then this is use-after-free.